src/components/ArticlePage.jsx

```diff
@@ -2,20 +2,47 @@ import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import { fetchArticle } from "../requests";
 import { Comments } from "./Comments";
+import { patchArticleLike } from "../requests";
 
 export function ArticlePage() {
   const { article_id } = useParams();
 
   const [articleData, setArticleData] = useState({});
   const [isLoading, setIsLoading] = useState(true);
+  const [likes, setLikes] = useState(0);
+  const [canLike, setCanlike] = useState(true);
+  const [likeText, setLikeText] = useState("Like");
 
   useEffect(() => {
     fetchArticle(article_id).then(({ data: { articles } }) => {
       setArticleData(articles);
       setIsLoading(false);
+      setLikes(articles.votes);
     });
   }, []);
 
+  const handleLike = () => {
+    if (canLike === true) {
+      setCanlike(false);
+      setLikes(likes + 1);
```

Comment:
Although this works the correct and failsafe way to get hold of the current value of state is to pass the state setting function a callback. This call back is invoked with the current state value guaranteeing its value is correct. Have a look [here](https://l2c.northcoders.com/courses/sd-notes/front-end#sectionId=optimistic-rendering,step=intro) for a an example 👍